hazelcast-client-legacy/src/main/java/com/hazelcast/client/proxy/NearCachedClientMapProxy.java

```diff
@@ -235,24 +237,36 @@ protected void loadAllInternal(boolean replaceExistingValues, List<Data> dataKey
     }
 
     @Override
-    protected MapEntries getAllInternal(List<Data> keys, Map<K, V> result) {
-        Iterator<Data> iterator = keys.iterator();
-        while (iterator.hasNext()) {
-            Data key = iterator.next();
-            Object cached = nearCache.get(key);
-            if (cached != null && NULL_OBJECT != cached) {
-                result.put((K) toObject(key), (V) cached);
-                iterator.remove();
+    protected List<MapEntries> getAllInternal(Map<Integer, List<Data>> partitionToKeyData, Map<K, V> result) {
+        List<Integer> partitionsWithCachedEntries = new ArrayList<Integer>(partitionToKeyData.size());
+        for (Entry<Integer, List<Data>> partitionKeyEntry : partitionToKeyData.entrySet()) {
+            List<Data> keyList = partitionKeyEntry.getValue();
+            Iterator<Data> iterator = keyList.iterator();
+            while (iterator.hasNext()) {
+                Data key = iterator.next();
+                Object cached = nearCache.get(key);
+                if (cached != null && NULL_OBJECT != cached) {
+                    result.put((K) toObject(key), (V) cached);
+                    iterator.remove();
+                }
+            }
+            if (keyList.isEmpty()) {
```

Comment:
I think this method could be simplified by leaving it as before and just adding a list.isEmpty() check on the getAllInternal call in the map. So you iterate over the map once, remove the cached entries, and then call the method on the base class. The super.getAllInternal will ensure that it doesn't make any requests for partitions with no keys.